plasma-mode.md

```diff
@@ -0,0 +1,203 @@
+```solidity
+enum ChallengeStatus {
+    Uninitialized, // Never emitted
+    Active,        // Emitted when a challenge is initiated
+    Resolved,      // Emitted when a challenge is resolved. This means the data has been made available on L1.
+    Expired        // Never emitted
+}
+
+event ChallengeStatusChanged(bytes indexed challengedCommitment, uint256 indexed challengedBlockNumber, ChallengeStatus status);
+```
+
+The flow is that after a commitment is posted to L1, it can be challenged within `challenge_window` blocks.
+If it is challenged an event with the `Active` status is emitted.
+If the challenge has not been resolved within `resolve_window` blocks, the challenge is presumed to suceed and the commitment is invalid.
+If the challenge is resolved within `resolve_window`, an event with the `Resolved` status is emitted and the commitment is valid.
+If a commitment is never challenged within `challenge_window` blocks, it is valid.
+
+This works essentially as an optimistic bridge. The commitment is assumed to be valid until someone claims that it is not.
+The validity is checked by a third party smart contract which understands the commitment format.
+
+
+#### V2: ZK Bridge
+
+DA Layers which have ZK bridges from their DA Chain to Ethereum can write a SC which emits the following event.
+
+```solidity
+    event MarkedInvalid(bytes indexed challengedCommitment, uint256 indexed challengedBlockNumber)
+    event MarkedValid(bytes indexed challengedCommitment, uint256 indexed challengedBlockNumber)
+```
+
+The flow is that after a commitment is posted to L1, it can be challenged within `challenge_window` blocks.
+If the challenge succeeds, the `MarkedInvalid` event is emitted. The `MarkedInvalid` event should not be emitted unless the data is not available for all nodes.
+If the challenge fails, the `MarkedValid` event is emitted. If the `MarkedValid` event is emitted, the data MUST be available for all nodes.
+
+### Derivation Integration
+
+The op-node uses the following interface inside the derivation pipeline.
+```go
+type CommStatus int
+const (
+    OptimisticOK     = iota
+    ConfirmedValid
+    ConfirmedInvalid
+)
+func CommitmentStatus(comm GenericCommitment, L1Source eth.BlockID) (CommStatus, error)
+```
+
+Commitment Status Defintions:
+- `OptimisticOK` means that the commitment is assumed to be valid, but could be challenged & invalidated later
+- `ConfirmedValid` means that the commitment cannot be invalidated on the given L1 chain
+- `ConfirmedInvalid` means that the commitment has been invalited & will not become valid on the given L1 chain
+
+We adapt the SC events into this golang interface to support multiple different bridge types without having to modify derivation.
+In addition, this confirmation model is more closely aligned to how the derivation pipeline works with commitment validity.
+
+When integrating with derivation, safe blocks are derived from commitments that are OptimisiticOK.
```

Comment:
This + Proto's doc on Derivation architecture works nicely -- when blocks become OptimisticOK, the DA Deriver can emit the SafeBlock event. When they are ConfirmedInvalid, it emits a Reorg event, and so on. Super cool.